### ISSUE TITLE
`switch-case-braces.js`: Please change the default to `avoid`

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#2415]
### Description

I've come to dislike `switch` because it's become so noisy and verbose over plain if/else, even if I have to repeat the condition multiple times.

This:

```js
if (event.key === 'Tab') {
	showExtras = true;
} else if (event.key === 'ArrowDown') {
	focusNext('.ext-name, [type="search"]');
} else if (event.key === 'ArrowUp') {
	focusPrevious('.ext-name, [type="search"]');
} else {
	return;
}
```

Turns into:

```js
switch (event.key) {
	case 'Tab': {
		showExtras = true;
		break;
	}

	case 'ArrowDown': {
		focusNext('.ext-name, [type="search"]');
		break;
	}

	case 'ArrowUp': {
		focusPrevious('.ext-name, [type="search"]');
		break;
	}

	default: {
		return;
	}
}
```

The latter is arguably more readable/airy, but so is this:

```js
switch (event.key) {
	case 'Tab':
		showExtras = true;
		break;

	case 'ArrowDown':
		focusNext('.ext-name, [type="search"]');
		break;

	case 'ArrowUp':
		focusPrevious('.ext-name, [type="search"]');
		break;

	default:
		return;
}
```

I don’t think that the braces add anything here.

### Fail

```js
var replace = 'me';
```

```js
function foo() {
	var replace = 'me';
	return replace;
}
```

### Pass

```js
const replace = 'me';
```

```js
function foo() {
	return 'me';
}
```

**Comment by fisker** [comment]
I use `avoid` too, but Sindre prefer `alway` https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1709#issuecomment-1030263041

**Comment by sindresorhus** [comment]
It's done for consistency. I feel it would look weird and inconsistent to only have braces when you need to use `const` inside a case.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#lexical_scoping

**Comment by sindresorhus** [comment]
I’m going to keep the default as `always`.

The rule is intentionaly opinionated around consistency. `switch` cases share a scope, so braces are needed as soon as a case declares a lexical binding. Having some cases braced and others not makes the style depend on implementation details inside each case, which is exactly the inconsistency this rule is meant to avoid.

I agree `avoid` can look cleaner in small examples, but changing the default would make the rule less consistent and would be a breaking style change for existing users. The current behavior remains the intended default, and projects that prefer the leaner style can configure the rule to `avoid`.
